graphics/src/ColladaLoader.cc

```diff
@@ -1467,38 +1482,116 @@ void ColladaLoader::Implementation::LoadPositions(const std::string &_id,
     if (count)
     {
       gzerr << "Vertex source missing float_array element, "
-        << "or count is invalid.\n";
+            << "or count is invalid.\n";
     }
     else
     {
       gzlog << "Vertex source has a float_array with a count of zero. "
-        << "This is likely not desired\n";
+            << "This is likely not desired\n";
     }
 
     return;
   }
+  // Read in the total number of position coordinate values
+  else if (floatArrayXml->Attribute("count"))
+    totCount = std::stoi(floatArrayXml->Attribute("count"));
+  else
+  {
+    gzerr << "<float_array> has no count attribute in position coordinate "
+          << "element with id[" << _id << "]\n";
+    return;
+  }
+
+  sourceXml = sourceXml->FirstChildElement("technique_common");
+  if (!sourceXml)
+  {
+    gzerr << "Unable to find technique_common element for texture "
+          << "coordinates with id[" << _id << "]\n";
+    return;
+  }
+
+  // Get the accessor XML element.
+  sourceXml = sourceXml->FirstChildElement("accessor");
+  if (!sourceXml)
+  {
+    gzerr << "Unable to find <accessor> as a child of <technique_common> "
+          << "for texture coordinates with id[" << _id << "]\n";
+    return;
+  }
+
+  // Read in the stride for the texture coordinate values. The stride
+  // indicates the number of values in the float array the comprise
+  // a complete texture coordinate.
+  if (sourceXml->Attribute("stride"))
+  {
+    stride = std::stoi(sourceXml->Attribute("stride"));
+  }
+  else
+  {
+    gzerr << "<accessor> has no stride attribute in texture coordinate "
+          << "element with id[" << _id << "]\n";
+    return;
+  }
+
+  // Nothing to read. Don't print a warning because the collada file is
+  // correct.
+  if (totCount == 0)
+    return;
+
   std::string valueStr = floatArrayXml->GetText();
 
   std::unordered_map<gz::math::Vector3d,
-      unsigned int, Vector3Hash> unique;
+                     unsigned int, Vector3Hash>
+      unique;
 
-  std::vector<std::string>::iterator iter, end;
-  std::vector<std::string> strs = split(valueStr, " \t\r\n");
-  end = strs.end();
-  for (iter = strs.begin(); iter != end; iter += 3)
+  auto to_double_vec = [](std::string_view sv, size_t total_count)
```

Comment:
Nit: We prefer camelStyle variables. Could you make a pass updating any other snake_case variables in your patch?